### PR TITLE
Downgrade use of lodash package to a weak dependency

### PR DESCRIPTION
**Pull request by rjsmith**
Downgraded use of eraser:meteor-lodash to a weak dependency in
easy:core and easy:components.
See #344

### CODE REVIEW

**Comment by rjsmith** [comment]
Hi @matteodem , I fixed up the failing tests, found a remaining use of `_.assign` at the same time. Should be OK now.

File: packages/easysearch:core/tests/unit/core/index-tests.js [hunk 1]
```diff
@@ -15,14 +15,18 @@ Tinytest.addAsync('EasySearch - Unit - Core - Index', function (test, done) {
...
+    test.equal(options.search.limit, 200);
+    test.equal(options.search.skip, 20);
+    test.equal(options.search.props, {});
   };
 
   index.search('testSearch', { limit: 200, skip: 20 });
 
   index.config.engine.search = function (searchDefinition, options) {
     test.equal(searchDefinition, 'testSearch');
-    test.equal(options.search, { limit: 10, skip: 0, props: { custom: 'property' } });
```
Comment (matteodem) [inline packages/easysearch:core/tests/unit/core/index-tests.js]: please revert these test fixes as this doesn't ensure anymore that the options have unwanted properties set

File: packages/easysearch:core/package.js [hunk 2]
```diff
@@ -10,8 +10,8 @@ Package.onUse(function(api) {
   api.versionsFrom('1.2.0.1');
 
   // Dependencies
-  api.use(['check', 'ecmascript', 'mongo']);
-  api.use(['erasaur:meteor-lodash@3.10.0']);
+  api.use(['check', 'ecmascript', 'mongo', 'underscore']);
+  api.use(['erasaur:meteor-lodash@3.10.0'], {weak: true});
```
Comment (matteodem) [inline packages/easysearch:core/package.js]: spaces missing around the curly braces

**Comment by matteodem** [comment]
I'll merge the PR when the fixes are commited and publish it in 2.0.3, thanks!

**Comment by rjsmith** [comment]
OK, the problem is I cannot get the following tests to pass when tinytest runs it on the Server (Client execution of same tests is OK):

```javascript
  index.config.engine.search = function (searchDefinition, options) {
    test.equal(searchDefinition, 'testSearch');
    test.equal(options.search, { limit: 200, skip: 20, props: {} })
  };

  index.search('testSearch', { limit: 200, skip: 20 });

  index.config.engine.search = function (searchDefinition, options) {
    test.equal(searchDefinition, 'testSearch');
    test.equal(options.search, { props: { custom: 'property' }, limit: 10, skip: 0 })
    done();
  };

  index.search('testSearch', { props: { custom: 'property' } });
```

Driving me nuts, any insight would be appreciated.

```
FAIL 10 ms
S: Index
- ok (4 times)
- fail — assert_equal - expected {"limit":200,"skip":20,"props":{}} - actual {"limit":200,"skip":20,"props":{}} - not
- ok
- fail — assert_equal - expected {"props":{"custom":"property"},"limit":10,"skip":0} - actual {"props":{"custom":"property"},"limit":10,"skip":0} - not
```

**Comment by rjsmith** [comment]
I think I figured out the problem, it was the use of `this.config.defaultSearchOptions` (without turning it into a function call) in the line below which seemed to screwing up the server-side execution of the test

```javascript
    this.defaultSearchOptions = _.defaults({}, this.config.defaultSearchOptions(), { limit: 10, skip: 0, props: {} });
```

**Comment by matteodem** [comment]
Is there now underscore equivalent for `_.defaults`?

**Comment by rjsmith** [comment]
I'll take a look at the merge conflicts.
AFAIK, _.defaults has been in underscore for at least 3 years, eg:

https://lostechies.com/chrismissal/2012/10/05/extending-objects-in-underscore/

**Comment by matteodem** [comment]
Thanks a lot! Gonna release this in 2.0.3

**Pull request merged**